Cargo.toml

```diff
@@ -82,6 +82,7 @@ console = "0.15.0"
 data-encoding = "2.2.0"
 enum-as-inner = "0.6"
 idna = "0.4.0"
+interfaces="0.0.8"
```

Comment:
This is pulling in a lot of dependencies for its build script, so I don't think this is an appropriate dependency to pull in for -proto by default.

https://crates.io/crates/network-interface seems like it's a bit more lightweight, although it still has a build script that depends on cc.

Reply:
Unfortunately, that crate doesn't give us an API for MTU: https://docs.rs/network-interface/latest/network_interface/struct.NetworkInterface.html.

Alternatively, I could just extract the value from the command line (e.g. `ifconfig`)

Reply:
Doing a cross-platform implementation of extracting the MTU from a command-line call doesn't exactly seem like something that belongs in this crate, either (and it still seems relatively inefficient). Maybe we should make the MTU configurable?